### PR TITLE
feat(PX-5269): Adds banner and FAQ screen for privately listed artworks

### PR DESCRIPTION
**Pull request by jpotts244**
This PR resolves [PX-5269](https://artsyproduct.atlassian.net/browse/PX-5269)

### Description

Display banner for "private artworks listings", aka artworks that have a `visibilityLevel === "UNLISTED"`
Banner text will link to a new info screen with additional details.

With help and pairing from @JanaeHijaz && @pvinis 
This is my first eigen PR! Feedback most welcome!!!!!! ✨ 
The initial MVP for the private artworks launch will be a beta release with only a handful of users so there will not be many artworks that will be marked as unlisted/private

https://user-images.githubusercontent.com/12748344/198124692-41f815e1-69de-43b5-a1e7-963c5e0f7c24.mov

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [x] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].
- [ ] I have prefixed changes that need to be tested during a release QA with **[NEEDS EXTERNAL QA]** on the changelog.

### To the reviewers 👀

- [x] I would like **at least one** of the reviewers to run this PR on the simulator or device. (This is my first eigen PR so would love some other eyes! I tested iOS but cannot get android simulator working)

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Adds banner and FAQ page for privately shared artworks @jpotts244

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/Artwork/Components/ArtworkHeader.tsx [hunk 1]
```diff
@@ -27,11 +27,18 @@ import { ArtworkActionsFragmentContainer as ArtworkActions, shareContent } from
...
 
 interface ArtworkHeaderProps {
   artwork: ArtworkHeader_artwork$data
 }
 
+export enum VisibilityLevels {
+  DRAFT = "DRAFT",
+  LISTED = "LISTED",
+  UNLISTED = "UNLISTED",
+}
```
Comment (jpotts244) [inline src/app/Scenes/Artwork/Components/ArtworkHeader.tsx:40]: The `visibilityLevel` is expected to be of type enum (coming from relay) so added a mapping here so we can pass the type checks

Not sure this is the best place for this to live, any thoughts?

File: src/app/AppRegistry.tsx [hunk 2]
```diff
@@ -441,6 +442,7 @@ export const modules = defineModules({
   Partner: reactModule(PartnerQueryRenderer),
   PartnerLocations: reactModule(PartnerLocations),
   PrivacyRequest: reactModule(PrivacyRequest),
+  PrivateListingsFAQScreen: reactModule(PrivateListingsFAQScreen),
   RequestForPriceEstimateScreen: reactModule(RequestForPriceEstimateScreen),
```
Comment (jpotts244) [inline src/app/AppRegistry.tsx:446]: CC @nspinazz89 reminder for us to sync on how to refer to this in the clients

Any thoughts PX-ers?

File: src/app/Scenes/Artwork/Components/PrivateListingsBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { navigate } from "app/navigation/navigate"
+import { Flex, LinkText, Text } from "palette"
+
+interface PrivateListingsBannerProps {
+  partnerName?: string | null
+}
+
+export const PrivateListingsBanner: React.FC<PrivateListingsBannerProps> = ({ partnerName }) => {
+  return (
+    <>
```
Comment (pvinis) [inline src/app/Scenes/Artwork/Components/PrivateListingsBanner.tsx:10]: non-blocking, i guess that fragment is not needed?

File: src/app/Scenes/Artwork/Components/ArtworkHeader.tests.tsx [hunk 4]
```diff
@@ -13,17 +14,50 @@ const TestRenderer: React.FC = () => {
 
 describe("ArtworkHeader", () => {
   it("renders tombstone component", () => {
-    const tree = renderWithWrappersLEGACY(<TestRenderer />)
-    expect(tree.root.findAllByType(ArtworkTombstone)).toHaveLength(1)
+    const { container } = renderWithWrappers(<TestRenderer />)
```
Comment (pvinis) [inline src/app/Scenes/Artwork/Components/ArtworkHeader.tests.tsx:17]: oh snap, nice!

**Review by pvinis — APPROVED** [review]

File: src/app/AppRegistry.tsx (hunk 2 above)
Comment (jpotts244) [reply src/app/AppRegistry.tsx:446]: ++ will adjust

File: src/app/Scenes/Artwork/Components/PrivateListingsBanner.tsx (hunk 3 above)
Comment (jpotts244) [reply src/app/Scenes/Artwork/Components/PrivateListingsBanner.tsx:10]: 👍🏻 will adjust! nice catch

File: src/app/Scenes/Artwork/Components/ArtworkHeader.tests.tsx (hunk 4 above)
Comment (jpotts244) [reply src/app/Scenes/Artwork/Components/ArtworkHeader.tests.tsx:17]: Hehe, lots of [doc](https://callstack.github.io/react-native-testing-library/docs/api-queries) reading 🤓 
but I think this captures the same behavior!

File: src/app/AppRegistry.tsx (hunk 2 above)
Comment (nspinazz89) [reply src/app/AppRegistry.tsx]: yeah I think Leandro makes a good point, we will use unlisted everywhere in the codebase and then adjust the copy for users as needed.

File: src/app/Scenes/Artwork/Components/UnlistedArtworksBanner.tests.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+describe("UnlistedArtworksBanner Screen", () => {
+  describe("when partner name present", () => {
+    it("renders the correct text", () => {
+      const { getByTestId } = renderWithWrappers(
+        <UnlistedArtworksBanner partnerName="Hello Partner" />
+      )
+      expect(getByTestId("unlisted-artworks-banner")).toHaveTextContent(
+        "This is a private listing from Hello Partner."
+      )
```
Comment (nspinazz89) [inline src/app/Scenes/Artwork/Components/UnlistedArtworksBanner.tests.tsx:12]: non blocking: A nice to have would be a test that confirms the modal/info screen text is there when you click "private listing"

**Review by nspinazz89 — APPROVED** [review]
Not an iOS expert, but this looks good to me, just one minor non-blocking comment.

**Review by gkartalis — APPROVED** [review]

**Pull request merged**